news/_posts/2021-06-15-multistore-news-in-1.7.8.md

```diff
@@ -0,0 +1,173 @@
+
+**Search bar** 
+
+
+In the specific settings drop-down, there is a search bar with auto-completion allowing you to search and select a shop or a group quickly. If the user chooses one of the listed results, it is automatically selected as context.
+
+In the specific settings drop-down there are:
+- The shop groups’ names with a link to select them as context
+- The shops’ names with a link to select them as context
+
+Next to each shop name, there is a mention:
+- "Inherited" if the parameter has the same value as in the “all shops” context
+- "Customized" if the parameter has a different value in a specific shop
+
+![Multistore specific settings dropdown](/assets/images/2021/06/multistore_specific_settings_dropdown.png)
+ 
+## IV.  Information messages
+Many information messages have been added on different pages and in different locations of the back office to provide better assistance, to inform of some behaviors as early as possible, and to avoid misunderstandings.
+
+For example, some pages are available only in the “all shops” context, and consequently, when you arrive on these pages, the context automatically switches to “all shops”. To prevent any issue, an information message has been added: "Note that this page is available in all shops context only, this is why your context has just switched."
+ 
+## V.  Modules
+Another pain identified during the user research step, was the difficulty of knowing whether a module is compatible with the multistore or not.
+
+To solve this problem the first thing to do was to define the notion of compatibility. A module can be :
+- compatible, 
+- partially compatible, 
+- incompatible 
+- or not concerned with the multistore compatibility.
+
+A module is considered compatible with the multistore feature when it can be either:
+- configured differently from one store to another;
+- configured quickly in the same way on all stores thanks to the “all shops context” or the group of shops;
+- or even activated for one store and deactivated for another.
+
+A partially compatible module means that some of its options might not be available or working in a multistore context.
+
+An incompatible module means that its configuration applies to all stores.
+
+A module may also not be concerned with multistore compatibility. For example, a captcha module will have the same configuration for all stores.
+
+
+The second thing was to indicate for each module their compatibility or not with the multistore. 
+
+For native modules’, we have indicated the multistore compatibility in the  [readme.md](https://github.com/PrestaShop/ps_emailalerts/pull/44/files) of each module.
+
+Modules developers can now indicate the multistore compatibility in the source code of their module, by calling the `setMultistoreCompatibility` method with one of the following constant values: `MULTISTORE_COMPATIBILITY_YES`, `MULTISTORE_COMPATIBILITY_PARTIAL`, `MULTISTORE_COMPATIBILITY_NO`, or `MULTISTORE_COMPATIBILITY_NOT_CONCERNED` 
```

Comment:
This has changed, there is no `setMultistoreCompatibility` method anymore.

The module developer must declare a `multistoreCompatibility` attribute, and give it one of the constants as a value, example:

`public $multistoreCompatibility = self:: MULTISTORE_COMPATIBILITY_YES`